sdk/nodejs/runtime/closure/codePaths.ts

```diff
@@ -106,52 +106,55 @@ export async function computeCodePaths(
-    const pathSet = await allFoldersForPackages(
+    // Find folders for all packages requested by the user.  Note: all paths in this should
+    // be normalized.
+    const normalizedPathSet = await allFoldersForPackages(
         new Set<string>(options.extraIncludePackages || []),
         new Set<string>(options.extraExcludePackages || []),
         options.logResource);
 
     // Add all paths explicitly requested by the user
     const extraIncludePaths = options.extraIncludePaths || [];
     for (const path of extraIncludePaths) {
-        pathSet.add(path);
+        // Note: use 'normalizeSafe' so that we don't unintentionally strip off "./" from the
+        // beginning in case that's what we were passed in.
+        normalizedPathSet.add(upath.normalizeSafe(path));
     }
 
     const codePaths: Map<string, asset.Asset | asset.Archive> = new Map();
 
     // For each of the required paths, add the corresponding FileArchive or FileAsset to the
     // AssetMap.
-    for (const path of pathSet) {
+    for (const normalizedPath of normalizedPathSet) {
         // Don't include a path if there is another path higher up that will include this one.
-        if (isSubsumedByHigherPath(path, pathSet)) {
+        if (isSubsumedByHigherPath(normalizedPath, normalizedPathSet)) {
             continue;
         }
 
         // The Asset model does not support a consistent way to embed a file-or-directory into an
         // `AssetArchive`, so we stat the path to figure out which it is and use the appropriate
         // Asset constructor.
-        const stats = fs.lstatSync(path);
+        const stats = fs.lstatSync(normalizedPath);
         if (stats.isDirectory()) {
-            codePaths.set(path, new asset.FileArchive(path));
+            codePaths.set(normalizedPath, new asset.FileArchive(normalizedPath));
         }
         else {
-            codePaths.set(path, new asset.FileAsset(path));
+            codePaths.set(normalizedPath, new asset.FileAsset(normalizedPath));
         }
     }
 
     return codePaths;
 }
 
-function isSubsumedByHigherPath(path: string, pathSet: Set<string>): boolean {
-    for (const otherPath of pathSet) {
-        if (path.length > otherPath.length &&
-            path.startsWith(otherPath)) {
+function isSubsumedByHigherPath(normalizedPath: string, normalizedPathSet: Set<string>): boolean {
+    for (const otherNormalizedPath of normalizedPathSet) {
+        if (normalizedPath.length > otherNormalizedPath.length &&
+            normalizedPath.startsWith(otherNormalizedPath)) {
 
             // Have to make sure we're actually a sub-directory of that other path.  For example,
             // if we have:  node_modules/mime-types, that's not subsumed by node_modules/mime
-            const nextChar = path.charAt(otherPath.length);
-            return nextChar === "/" || nextChar === "\\";
+            const nextChar = normalizedPath.charAt(otherNormalizedPath.length);
+            return nextChar === "/";
```

Comment:
@swgillespie What do you think about this helper function?  Should it actually do a split to get the path components and then check those?  Or is this code clear/safe enough for you?

Reply:
>  Should it actually do a split to get the path components and then check those?

I think in general I'd prefer that. I believe that this code is correct, but I have to squint a little to prove to myself that the `startsWith` logic is correct. I'm fine with how it is now, but if we keep running into issues or need to re-write this for whatever reason it would be nice to stop treating paths as raw strings.

Don't block on that though, I think this logic is correct as long as we assume that `upath` isn't buggy, but filepaths are so complicated I can't help but wonder if there's weird edge-case bugs lurking.

Reply:
ok.  will leave like this for now. main concern if i switched would just be if thigns got really bad if there were a ton of paths.  as it is right now, we're already N^2.  Adding splitting/walking blows that up more and i'd be worried about then needing to go and make this efficient.

Reply:
sounds good to me.